reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -255,4 +252,86 @@ promise_test(() => {
   return writer.abort().then(() => assert_true(thenCalled, 'then() should be called'));
 }, 'returning a thenable from abort() should work');
 
+promise_test(t => {
+  const ws = new WritableStream({
+    write() {
+      return flushAsyncEvents();
+    }
+  });
+  const writer = ws.getWriter();
+  return writer.ready.then(() => {
+    const writePromise = writer.write('a');
+    writer.abort(new Error());
+    let closedResolved = false;
+    return Promise.all([
+      writePromise.then(() => assert_false(closedResolved, '.closed should not resolve before write()')),
+      promise_rejects(t, new TypeError(), writer.closed, '.closed should reject')
+          .then(() => {
+            closedResolved = true;
+          })]);
+  });
+}, '.closed should not resolve before write()');
+
+promise_test(t => {
+  const ws = new WritableStream({
+    write() {
+      return Promise.reject(new Error());
+    }
+  });
+  const writer = ws.getWriter();
+  return writer.ready.then(() => {
+    const writePromise = writer.write('a');
+    writer.abort(new Error());
+    let closedResolved = false;
+    return Promise.all([
+      promise_rejects(t, new Error(), writePromise, 'write() should reject')
+          .then(() => assert_false(closedResolved, '.closed should not resolve before write()')),
+      promise_rejects(t, new TypeError(), writer.closed, '.closed should reject')
```

Comment:
Okay, this is done. It should get simpler if we make write() and close() uninterruptable as discussed in #617.